src/Google/RequestReviewStatuses.php

```diff
@@ -11,79 +11,96 @@ class RequestReviewStatuses implements Service {
 
-					$issues = array_merge( $issues, $region_status['reviewIssues'] ?? [] );
-				}
+		$valid_program_states = [ 'ENABLED', self::NO_OFFERS ];
 
-				if ( $status === self::DISAPPROVED ) {
-					continue;
-				}
-
-				if ( $region_status['eligibilityStatus'] === self::UNDER_REVIEW ) {
-					$status = self::UNDER_REVIEW;
-				}
-
-				if ( $status === self::UNDER_REVIEW ) {
-					continue;
-				}
+		foreach ( $response as $program_type ) {
 
-				if ( $region_status['eligibilityStatus'] === self::PENDING_REVIEW ) {
-					$status = self::PENDING_REVIEW;
-				}
+			// In case any Program is with no offers we consider it Onboarding
+			if ( $program_type['data']['globalState'] === self::NO_OFFERS ) {
+				$status = self::ONBOARDING;
+				break;
+			}
 
-				if ( $status === self::PENDING_REVIEW ) {
-					continue;
-				}
+			// In case any Program is not enabled or there are no regionStatuses we return null status
+			if ( ! isset( $program_type['data']['regionStatuses'] ) || ! in_array( $program_type['data']['globalState'], $valid_program_states, true ) ) {
+				$status = null;
+				break;
+			}
 
-				if ( $region_status['eligibilityStatus'] === self::ONBOARDING ) {
-					$status = self::ONBOARDING;
-				}
+			// otherwise we compute the new status, issues and cooldown period
+			foreach ( $program_type['data']['regionStatuses'] as $region_status ) {
+				$issues = array_merge( $issues, $region_status['reviewIssues'] ?? [] );
+				self::maybe_update_cooldown_period( $region_status, $cooldown );
+				self::maybe_update_status( $region_status['eligibilityStatus'], $status );
 			}
 		}
 
 		return [
-			'issues'   => array_map( 'strtolower', array_unique( $issues ) ),
+			'issues'   => array_map( 'strtolower', array_values( array_unique( $issues ) ) ),
 			'cooldown' => self::get_cooldown( $cooldown ), // add lifetime cache to cooldown time
 			'status'   => $status,
 		];
 	}
+	/**
+	 * Updates the cooldown period in case the new cooldown period date is available and later than the current cooldown period.
+	 *
+	 * @param array $region_status Associative array containing (maybe) a cooldown date property.
+	 * @param int   $cooldown Referenced current cooldown to compare with
+	 */
+	private function maybe_update_cooldown_period( $region_status, &$cooldown ) {
+		if (
+			isset( $region_status['reviewIneligibilityReasonDetails'] ) &&
+			isset( $region_status['reviewIneligibilityReasonDetails']['cooldownTime'] )
+		) {
+			$region_cooldown = intval( strtotime( $region_status['reviewIneligibilityReasonDetails']['cooldownTime'] ) );
+
+			if ( ! $cooldown || $region_cooldown > $cooldown ) {
+				$cooldown = $region_cooldown;
+			}
+		}
+	}
+
+	/**
+	 * Updates the status reference in case the new status has more priority.
+	 *
+	 * @param String $new_status New status to check has more priority than the current one
+	 * @param String $status Referenced current status
+	 */
+	private function maybe_update_status( $new_status, &$status ) {
```

Comment:
Because we are modifying the variable status without doing the equivalent `$status = maybe_update_status($new_status, $status);` 

https://www.php.net/manual/en/language.references.pass.php

Reply:
Thanks, however, I do not see the benefit in this case to pass-by-reference instead of pass-by-value. Using the pass-by-reference makes the code more susceptible to creating bugs as you are updating variables outside of the function scope. Looking into the codebase you see that mostly all the cases that pass-by-reference are used, it is when we are updating the actual values of an array, in that case, I think it makes more sense to use the reference. 

Example: 

https://github.com/woocommerce/google-listings-and-ads/blob/fb70c94c21037c37352ee7d8d30d90362b1d6fc4/src/API/Site/Controllers/MerchantCenter/IssuesController.php#L92

Reply:
I don't think passing a var by reference will create bugs. Also notice that both are private functions using internally. 

However, I update it to by-value to get the PR moving and make it clearer.